components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -5751,7 +5952,75 @@ def getBody(self):
 }
 %s
 vp.set(UndefinedValue());
-return true;""" % (getIndexedOrExpando, getNamed)
+return true;""" % (maybeCrossOriginGet, getIndexedOrExpando, getNamed)
+
+    def definition_body(self):
+        return CGGeneric(self.getBody())
+
+
+class CGDOMJSProxyHandler_set(CGAbstractExternMethod):
+    def __init__(self, descriptor):
+        args = [Argument('*mut JSContext', 'cx'), Argument('RawHandleObject', 'proxy'),
+                Argument('RawHandleId', 'id'), Argument('RawHandleValue', 'v'),
+                Argument('RawHandleValue', 'receiver'),
+                Argument('*mut ObjectOpResult', 'opresult')]
+        CGAbstractExternMethod.__init__(self, descriptor, "set", "bool", args)
+        self.descriptor = descriptor
+
+    def getBody(self):
+        descriptor = self.descriptor
+
+        # `CGDOMJSProxyHandler_set` doesn't support legacy platform objects'
+        # `[[Set]]` (https://heycam.github.io/webidl/#legacy-platform-object-set) yet.
+        #
+        assert descriptor.isMaybeCrossOriginObject()
+        assert not descriptor.operations['IndexedGetter']
+        assert not descriptor.operations['NamedGetter']
+
+        maybeCrossOriginSet = dedent(
+            """
+            if !proxyhandler::is_platform_object_same_origin(cx, proxy) {
+                return proxyhandler::cross_origin_set(cx, proxy, id, v, receiver, opresult);
+            }
+
+            // Safe to enter the Realm of proxy now.
+            let _ac = JSAutoRealm::new(*cx, proxy.get());
+            """)
+
+        return dedent(
+            """
+            let cx = SafeJSContext::from_ptr(cx);
+            %(maybeCrossOriginSet)s
+
+            // OrdinarySet
+            // <https://tc39.es/ecma262/#sec-ordinaryset>
+            rooted!(in(*cx) let mut own_desc = PropertyDescriptor::default());
+            if !getOwnPropertyDescriptor(*cx, proxy, id, own_desc.handle_mut().into()) {
+                return false;
+            }
+
+            js::jsapi::SetPropertyIgnoringNamedGetter(
+                *cx, proxy, id, v, receiver, own_desc.handle().into(), opresult)
+            """) % {"maybeCrossOriginSet": maybeCrossOriginSet}
+
+    def definition_body(self):
+        return CGGeneric(self.getBody())
+
+
+class CGDOMJSProxyHandler_getPrototype(CGAbstractExternMethod):
```

Comment:
Same here - it looks like this could be a common function in dom/bindings/proxyhandler.rs instead.

Reply:
Addressed in 591ce9e4168738c4bdac0ecbd8e9155101644ba2

Reply:
Nevermind - `CGDOMJSProxyHandler_getPrototype` actually requires a function pointer to an interface-specific `GetProtoObject` implementation.